examples/swap_networks.py

```diff
@@ -0,0 +1,83 @@
+"""Demonstrates swap networks."""
+
+import itertools
+import random
+
+import cirq
+import cirq.contrib.acquaintance as cca
+
+
+def get_random_graph(n_vertices, edge_prob=0.5):
+    return [
+        ij for ij in itertools.combinations(range(n_vertices), 2)
+        if random.random() <= edge_prob
+    ]
+
+
+def get_max_cut_qaoa_circuit(n_vertices, edges, beta, gamma):
+    """Implement a single round of QAOA for MaxCut using linearly connected
+    qubits.
+
+    Args:
+        n_vertices: The number of vertices.
+        edges: The list of edges.
+        beta: The mixing angle.
+        gamma: The phase separation angle.
+
+    Returns:
+        A circuit implementing a single round of QAOA with the specified
+        angles.
+    """
+
+    # G_{i,j} ∝ exp(i gamma (|01><01| + |10><10|))
+    phase_sep_gates = {edge: cirq.ZZ**gamma for edge in edges}
```

Comment:
So I was just playing around with this code and I want to point out what I view as a somewhat insidious trap I ran into. Basically, I am using a different data structure for representing the graphs I want to simulate and so for various reasons, it was natural for me to keep track of edges in a slightly different way. I built this dictionary using keys that represented edges between qubits and looked like (cirq.LineQubit(i), cirq.LineQubit(j)) as opposed to edges that looked like (i, j). This did not raise an error when I called the GreedyExecutionStrategy. Instead, it just ignored the gates I was trying to insert. Instead, I think it should raise an exception that says something like "error, your edges look wrong". Let me know if what I'm saying is clear and if you agree. To be clear, I think the problem here is more with the code that silently ignores dictionary entries that must live in GreedyExecutionStrategy, not in the demo per se.

Reply:
One tool that could help here is type annotations. They inform whoever reads or modifies the code how vertices, edges etc are represented here. See comment above.

Reply:
@babbush It's actually perfectly fine to use keys that are tuples of qubits rather than integers; the initial mapping just needs to do that as well. The example code now shows both ways.

@viathor Type annotations added. These are actually just copied and pasted from cirq/contrib/acquaintance/permutation.py, but that seemed more concise than importing them and then having their definitions in a comment.